lightwood/analysis/model_analyzer.py

```diff
@@ -130,11 +131,12 @@ def model_analyzer(
         elif is_multi_ts:
             # we fit ICPs for time series confidence bounds only at t+1 forecast
             icp.nc_function.model.prediction_cache = np.array([p[0] for p in normal_predictions['prediction']])
-
         else:
             icp.nc_function.model.prediction_cache = np.array(normal_predictions['prediction'])
 
         if not is_classification:
+            data[target] = data[target].astype(float)
+            tr_data[target] = tr_data[target].astype(float)
```

Comment:
Why does this happen here and not in the data cleaner ?

Reply:
Agreed, will move to the data cleaner